### DISCUSSION TITLE
Map n fields to a collection (Many to One)

### DISCUSSION THREAD

**Discussion by grigoni** [mapstruct/mapstruct#2605] [Q&A]
hello

Object with fields -> iterable

I'm trying to map #n fields from source object to a CmisPropertiesType that contains `List<CmisProperty> property`
is there a way or a suggested workaround?

best regards
Gianandrea

**Comment by filiphr** [comment]
From what I understand you are trying to map the fields of one object to a collection of tupples.

This is not something that is supported by MapStruct. However, I do think that it is easy to write your own Annotation Processor that will do what you are looking for.

If I did not understand your question please provide an example with the objects and how you would do it by hand right now.

**Comment by grigoni** [comment]
your understanding is correct, I did a workaround with a factory. maybe I will try a custom Annotation Processor in the future, thanks!

**Comment by grigoni** [comment]
I'd be also very interested in bean to map, see https://github.com/mapstruct/mapstruct/issues/2504

**Comment by filiphr** [comment]
We are currently not focused on that feature (bean to map) @grigoni. If someone is willing to help out we are more than happy to guide them.
